app/controllers/settings_controller.rb

```diff
@@ -256,6 +256,25 @@ def twitter_disconnect
     return redirect_to "/settings"
   end
 
+  def keybase_auth
+    if !params[:kb_username].present? || !params[:kb_signature].present? || !params[:kb_ua].present? || (!params[:username].present? || !params[:username] == @user.username)
+      flash[:error] = "Invalid Keybase connection parameters"
+      return redirect_to "/settings"
+    end
+
+    kb_username = params[:kb_username]
+    kb_signature = params[:kb_signature]
+    kb_ua = params[:kb_ua]
+
+    if Keybase.validate_initial(kb_username, kb_signature, @user.username)
+      @user.keybase_signatures = [{ kb_username: kb_username, sig_hash: kb_signature }]
+      return redirect_to Keybase.success_url(kb_username, kb_signature, kb_ua, @user.username)
+    else
+      flash[:error] = "Failed to connect your account to Keybase, invalid proof signature was provided."
```

Comment:
What's next step can we give the user here? Should they be contacting Keybase about their bug, or mailing the site admin about our own bug?

Reply:
it's more of a 4xx than a 5xx, since invalid auth data was provided

Reply:
Can the user fix it? If they can, let's tell them what to change. If it's something in our json config that's borked, tell them to email the admin.

Reply:
it's likely that the only situation where the user can do anything is if they're logged in as a different user than the one they're trying to prove. other than that, it's (1) a problem in the request being made by lobsters (maybe catch this with logging and general observability), (2) an error in keybase (not much we can do about it here i'm sorry to say), (3) the user is accidentally trying to prove something that will never work (maybe a notification that they should try again from keybase), (4) the user is purposefully trying to prove something that will never work (fuck 'em).